server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequest.java

```diff
@@ -32,6 +34,7 @@ public class NodesStatsRequest extends BaseNodesRequest<NodesStatsRequest> {
 
     private CommonStatsFlags indices = new CommonStatsFlags();
     private final Set<String> requestedMetrics = new HashSet<>();
+    private NodeStatsLevel level = NodeStatsLevel.NODE;
```

Comment:
Hmm I don't think this makes sense as a field on the request (e.g. we're not sending it over the wire). This parameter only matters when sending back the response. I don't think we should add this field, we should just make sure to fail when generating the request in the REST layer if the parameter is invalid. (Similarly in other places that add request parameters which are not sent over the wire)